test/unit/gateways/global_collect_test.rb

```diff
@@ -71,6 +91,30 @@ def test_successful_purchase_with_requires_approval_true
     end.respond_with(successful_authorize_response, successful_capture_response)
   end
 
+  def test_purchase_request_with_google_pay
+    stub_comms do
+      @gateway.purchase(@accepted_amount, @google_pay_network_token)
+    end.check_request(skip_response: true) do |_endpoint, data, _headers|
+      assert_equal '320', JSON.parse(data)['mobilePaymentMethodSpecificInput']['paymentProductId']
+    end
+  end
+
+  def test_purchase_request_with_google_pay_pan_only
+    stub_comms do
+      @gateway.purchase(@accepted_amount, @google_pay_pan_only, @options.merge(customer: 'GP1234ID', google_pay_pan_only: true))
+    end.check_request(skip_response: true) do |_endpoint, data, _headers|
+      assert_equal '320', JSON.parse(data)['mobilePaymentMethodSpecificInput']['paymentProductId']
+    end
+  end
+
+  def test_purchase_request_with_apple_pay
+    stub_comms do
+      @gateway.purchase(@accepted_amount, @apple_pay_network_token)
+    end.check_request(skip_response: true) do |_endpoint, data, _headers|
+      assert_equal '302', JSON.parse(data)['mobilePaymentMethodSpecificInput']['paymentProductId']
+    end
+  end
+
```

Comment:
💬  Comment:

Given this PR has a lot of logic on the `add_decrypted_payment_data` and `add_order` methods, could be a good idea to test here different scenarios to ensure that "mobilePaymentMethodSpecificInput" or "cardPaymentMethodSpecificInput" gets populated properly

Reply:
Ok, then I'll add more tests related to those methods.